lib/much-slug/activerecord.rb

```diff
@@ -2,46 +2,52 @@
 require "much-slug"
 
 module MuchSlug
-
   module ActiveRecord
     include MuchPlugin
 
     plugin_included do
-      extend ClassMethods
-      include InstanceMethods
-
-      @much_slug_has_slug_configs = MuchSlug::Configs.new
+      @much_slug_has_slug_registry = MuchSlug::HasSlugRegistry.new
     end
 
-    module ClassMethods
-
-      def has_slug(options = nil)
-        options ||= {}
-        attribute = MuchSlug.set_has_slug(@much_slug_has_slug_configs, options)
+    plugin_class_methods do
+      def has_slug(
+            source:,
+            attribute: nil,
+            preprocessor: nil,
+            separator: nil,
+            allow_underscores: nil,
+            skip_unique_validation: false,
+            unique_scope: nil)
+        registered_attribute =
+          self.much_slug_has_slug_registry.register(
+            attribute: attribute,
+            source: source,
+            preprocessor: preprocessor,
+            separator: separator,
+            allow_underscores: allow_underscores,
+          )
 
         # since the slug isn't written until an after callback we can't always
         # validate presence of it
-        validates_presence_of(attribute, :on => :update)
+        validates_presence_of(registered_attribute, :on => :update)
 
-        if options[:skip_unique_validation] != true
-          validates_uniqueness_of(attribute, {
+        if skip_unique_validation != true
```

Comment:
@kellyredding - Do we want to just use `if skip_unique_validation` here? I guess it seems like specifically looking for `true` or `false` isn't necessary and we can rely on it being truthy or not.